src/api/docgen/transpile/python.ts

```diff
@@ -0,0 +1,318 @@
+    const type = this.type(callable.parentType);
+
+    const parameters = new Array<reflect.Parameter>();
+
+    for (const p of callable.parameters.sort(this.optionalityCompare)) {
+      if (!this.isStruct(p)) {
+        parameters.push(p);
+      } else {
+        // struct parameters are expanded to the individual struct properties
+        const struct = p.parentType.system.findInterface(p.type.fqn!);
+        for (const property of struct.allProperties) {
+          const parameter = propertyToParameter(callable, property);
+          parameters.push(parameter);
+        }
+      }
+    }
+
+    const name = toSnakeCase(callable.name);
+    const inputs = parameters.map((p) => formatInput(this.parameter(p)));
+
+    return {
+      name,
+      parentType: type,
+      import: formatImport(type),
+      parameters,
+      signature: formatSignature(name, inputs),
+      invocation: formatInvocation(
+        type,
+        inputs,
+        callable.kind === reflect.MemberKind.Initializer ? undefined : name
+      ),
+    };
+  }
+
+  public interface(
+    iface: reflect.InterfaceType
+  ): transpile.TranspiledInterface {
+    return {
+      name: iface.name,
+      type: this.type(iface),
+    };
+  }
+
+  public type(type: reflect.Type): transpile.TranspiledType {
+    const module = this.moduleFqn(type.fqn);
+
+    const fqn = [module];
+
+    if (type.namespace) {
+      fqn.push(type.namespace);
+    }
+
+    fqn.push(type.name);
+
+    return { fqn: fqn.join("."), moduleFqn: module };
+  }
+
+  private moduleFqn(fqn: string): string {
+    const type = this.ts.findFqn(fqn);
+    const submodule = this.findSubmodule(type);
+
+    const targets = submodule ? submodule.targets : type.assembly.targets;
+
+    if (!targets) {
+      throw new Error(`Unable to find 'targets' for fqn ${type.fqn}`);
+    }
+
+    if (!targets.python) {
+      throw new Error(`Python is not a supported target for fqn ${type.fqn}`);
+    }
+
+    return targets.python.module;
+  }
+
+  private optionalityCompare(
+    p1: reflect.Parameter,
+    p2: reflect.Parameter
+  ): number {
+    if (!p1.optional && p1.optional) {
+      return -1;
+    }
+    if (!p2.optional && p1.optional) {
+      return 1;
+    }
+    return 0;
+  }
+
+  private isStruct(p: reflect.Parameter): boolean {
+    return p.type.fqn ? p.system.findFqn(p.type.fqn).isDataType() : false;
+  }
+
+  private findSubmodule(type: reflect.Type): reflect.Submodule | undefined {
+    if (!type.namespace) {
+      return undefined;
+    }
+
+    // if the type is in a submodule, the submodule name is the first
+    // part of the namespace. we construct the full submodule fqn and seach for it.
+    const submoduleFqn = `${type.assembly.name}.${
+      type.namespace.split(".")[0]
+    }`;
+    const submodules = type.assembly.submodules.filter(
+      (s) => s.fqn === submoduleFqn
+    );
+
+    if (submodules.length > 1) {
```

Comment:
Added a comment. This can't happen. Its essentially just a side-effect of converting a dictionary of submodules (as stored in the spec) to an array (as exposed here).